### ISSUE TITLE
Helm throws an error when packaging charts

### ISSUE THREAD

**Issue by ursuad** [helm/helm#2258]
```
$ helm version
Client: &version.Version{SemVer:"v2.3.0", GitCommit:"2342275b61e0539d259590975cfb78f23afcc1e3", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.3.0", GitCommit:"d83c245fc324117885ed83afc90ac74afed271b4", GitTreeState:"clean"}
```

**Steps to reproduce the problem**:

* create a test chart
```
$ helm create test
```
* test that what you wanted to create is there and change to that directory:
```
$ cd test && tree .
.
├── Chart.yaml
├── charts
├── templates
│   ├── NOTES.txt
│   ├── _helpers.tpl
│   ├── deployment.yaml
│   └── service.yaml
└── values.yaml
```
* lint the chart:  
```
$ helm lint .
==> Linting .
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, no failures
```
* package the chart:
```
$ helm package .
```
**Expected result**:  Chart gets packaged in a `.tgz` file and the command returns `0`. 
**ACTUAL RESULT**: The `.tgz` file gets generated, but the command returns an error code:
```
$ helm package .
WARNING: Deprecated index file format. Try 'helm repo update'
Error: no API version specified
$ echo $?
1
```
 **Other steps taken**

* update the repo: 
```
$ helm repo update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "dp-stable" chart repository
Update Complete. ⎈ Happy Helming!⎈
$ helm package .
WARNING: Deprecated index file format. Try 'helm repo update'
Error: no API version specified
```

**Comment by ursuad** [comment]
**More info**
This happened after the helm upgrade to the latest version. 

I managed to fix this by:

* uninstalling helm with `brew uninstall kubernetes-helm`
* deleting the `~/.helm` directory
* reinstalling `kubernetes-helm` with brew which made it recreate all the directories. 

So probably this should be fixed as a part of the upgrade process, or some documentation?

**Comment by technosophos** [comment]
What version were you on prior to the update? The error displayed there is from a feature migration between Helm 2-Alpha.2 and Alpha.3. So if it's still showing up in a modern codebase, I'll need to do some digging.

**Comment by technosophos** [comment]
I've tested migrations from 2.2.0 and 2.3.0 and cannot reproduce.

For those of you looking to reproduce, the error in question is caused when $HELM_HOME/repositories/repositories.yaml does not have `apiVersion: v1` specified. It _could_ be that `helm init` was never run (or run with --skip-refresh), or that the repository was created with Helm 2.0.0-Alpha.2 or earlier. `helm repo update` fixes the problem by regenerating the repositories file if it detects an out-of-date version.

The warning that appears is correct behavior. So the question we need to identify is whether there was an incompatibility between Helm ??? and Helm 2.3.0.

**Comment by ursuad** [comment]
I upgraded from the previous release `2.2.3` and I didn't have Helm installed before that.

**Comment by technosophos** [comment]
Can you let me know if you can get it back into the broken state? There are two possible files that could cause the warning above. Both are created by `helm init`, but one gets updated during `helm package` and one gets updated by `helm serve`.

**Comment by ae6rt** [comment]
I am a new helm user as of yesterday.  I am running these versions

```
Client: &version.Version{SemVer:"v2.4.2", GitCommit:"82d8e9498d96535cc6787a6a9194a76161d29b4c", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.4.2", GitCommit:"82d8e9498d96535cc6787a6a9194a76161d29b4c", GitTreeState:"clean"}
```

I ran *helm init* without incident and additional flags.

My repositories.yaml file is

```
apiVersion: v1
generated: 2017-06-05T13:00:51.914671353-07:00
repositories:
- caFile: ""
  cache: kubernetes-charts-index.yaml
  certFile: ""
  keyFile: ""
  name: kubernetes-charts
  url: http://storage.googleapis.com/kubernetes-charts
```
When experimenting with packaging, I encounter messages similar to the OP:

```
$ helm package ./landru
WARNING: Deprecated index file format. Try 'helm repo update'
Error: no API version specified
```
though the archive appears to have been created successfully

```
$ tar tvf landru-1.0.5.tgz
-rwxr-xr-x  0 0      0         321 Dec 31  1969 landru/Chart.yaml
-rwxr-xr-x  0 0      0          51 Dec 31  1969 landru/values.yaml
-rwxr-xr-x  0 0      0        4462 Dec 31  1969 landru/templates/config.yaml
-rwxr-xr-x  0 0      0        1510 Dec 31  1969 landru/templates/deployment.yaml
-rwxr-xr-x  0 0      0         187 Dec 31  1969 landru/templates/eviction-budget.yaml
-rwxr-xr-x  0 0      0          89 Dec 31  1969 landru/templates/service-account.yaml
-rwxr-xr-x  0 0      0         215 Dec 31  1969 landru/templates/service.yaml
```
I have time to help debug with guidance.

**Comment by technosophos** [comment]
Can you run `helm serve`, @ae6rt ? Does it give you any errors?

One possible reason for this error is that `helm init` was never run (only `helm init --client-only` is necessary. Another is that `helm serve` corrupted the index. And often just running it again will fix the index. As soon as it has successfully started up, you can CTRL-C it.

**Comment by ae6rt** [comment]
Running "helm serve ." fixed the issue for me.

**Comment by bacongobbler** [comment]
closing as inactive, but please reopen if this is still an issue.

**Comment by wknapik** [comment]
I'm seeing this in my CI jobs from time to time.

At this point, due to #3111, I do `rm -fr ~/.helm` at the beginning of the pipeline ($HOME is per branch and only one pipeline can run for each branch at the same time, so there's no race condition).

This is what I saw today (this is output from a make target that calls `helm package`):
```
make: *** Deleting file '...'
Successfully packaged chart and saved it to: ...
WARNING: Deprecated index file format. Try 'helm repo update'
Error: no API version specified
Makefile:207: recipe for target ... failed
make: *** [...] Error 1
```
The output looks like it's out of order, but make was called with `-O`, so it shouldn't be. The "Deleting file" at the beginning is due to `.DELETE_ON_ERROR`.

This appears to happen randomly from time to time. Re-running the build succeeds. Seems like it might be related to the parallel execution of `helm package` targets ?

I've been seeing it with multiple helm versions. Today it was:
```
Client: &version.Version{SemVer:"v2.7.2", GitCommit:"8478fb4fc723885b155c924d1c8c410b7a9444e6", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.7.2", GitCommit:"8478fb4fc723885b155c924d1c8c410b7a9444e6", GitTreeState:"clean"}
```

**Comment by wknapik** [comment]
@bacongobbler 
(not sure if you get notifications for closed issues)

**Comment by bacongobbler** [comment]
I’ll re-open this one then. :)

**Comment by sgrigorev** [comment]
This issue happens for me if I run `$ helm dependencies update` in parallel for several charts.

**Comment by technosophos** [comment]
@sgrigorev Can you please open a separate issue to track that one? I think the cause may be different when doing multiple dep up operations. And can you include the usual Helm/Kube version info there?

**Comment by sgrigorev** [comment]
@technosophos, I've created the issue, see #4463

**Comment by bilalyasar** [comment]
this issue too annoying and we see most of the time.
@michelleN do you have any estimation for this issue? when it can be solved?

**Comment by bacongobbler** [comment]
Feel like working on a fix? :)

**Comment by bilalyasar** [comment]
this is really blocker issue sometimes. if you need any extra info let me know.

**Comment by wknapik** [comment]
If you're calling `helm package` from a makefile, this code seems to work around the problem. 

```make
# Use bash for recipes and exit with 1 on any non-zero exit.
SHELL := $(shell which bash)
.SHELLFLAGS := -eo pipefail -c

# Retry any task up to N times.
retry = for ((i=0; i < $1; ++i)); do ( $2 ) && exit 0; sleep 2; done; false

some_helm_package_target:
    $(call retry,2,rm -f "$@"; helm package ...)
```

If `helm package` fails, we sleep for 2s, delete the product of the failed run (if any) and try again.
Seems to work for me so far...

**Comment by bacongobbler** [comment]
closing as inactive. Please re-open with a new ticket if this persists with Helm 3.2.1+.
